src/frame/headers.rs

```diff
@@ -283,9 +283,86 @@ impl fmt::Debug for Headers {
     }
 }
 
+// ===== util =====
+
+pub fn parse_u64(src: &[u8]) -> Result<u64, ()> {
```

Comment:
Since this is used in multiple locations now (I believe), I would consider creating a private `util` module at the root and moving this there.